common/persistence/client/bean_test.go

```diff
@@ -0,0 +1,369 @@
+				},
+				err: "no history manager",
+			},
+			"config manager error": {
+				mockSetup: func(t *testing.T, f *MockFactory) {
+					f.EXPECT().NewConfigStoreManager().Return(nil, fmt.Errorf("no config manager"))
+				},
+				err: "no config manager",
+			},
+		}
+		for name, test := range tests {
+			name, test := name, test
+			t.Run(name, func(t *testing.T) {
+				t.Parallel()
+				f, _, defaultMocks := beanSetup(t)
+				test.mockSetup(t, f)
+				defaultMocks()
+				impl, err := NewBeanFromFactory(f, nil, nil)
+				if test.err != "" {
+					assert.ErrorContains(t, err, test.err)
+				} else {
+					assert.NoError(t, err)
+					assert.NotNil(t, impl)
+				}
+			})
+		}
+	})
+	t.Run("Simple getters", func(t *testing.T) {
+		t.Parallel()
+		f, m, defaultMocks := beanSetup(t)
+		defaultMocks() // build all mock objects
+		impl, err := NewBeanFromFactory(f, nil, nil)
+		require.NoError(t, err)
+
+		// these need to be concurrency-safe, so run them concurrently
+		var g errgroup.Group
+		g.Go(errgroupAssertEqual(t, m.domainManager, impl.GetDomainManager))
+		g.Go(errgroupAssertEqual(t, m.taskManager, impl.GetTaskManager))
+		g.Go(errgroupAssertEqual(t, m.visibilityManager, impl.GetVisibilityManager))
+		g.Go(errgroupAssertEqual(t, m.replicationManager, impl.GetDomainReplicationQueueManager))
+		g.Go(errgroupAssertEqual(t, m.shardManager, impl.GetShardManager))
+		g.Go(errgroupAssertEqual(t, m.historyManager, impl.GetHistoryManager))
+		g.Go(errgroupAssertEqual(t, m.configManager, impl.GetConfigStoreManager))
+		require.NoError(t, g.Wait())
+		// execution managers are per shard, checked separately
+	})
+	t.Run("Simple setters", func(t *testing.T) {
+		t.Parallel()
+		f, _, defaultMocks := beanSetup(t)
+		_, m2, _ := beanSetup(t) // make a second set of mock objects, so they can be compared
+		defaultMocks()           // allow constructors to be called
+		impl, err := NewBeanFromFactory(f, nil, nil)
+		require.NoError(t, err)
+
+		// these need to be concurrency-safe, so run them concurrently
+		var g errgroup.Group
+
+		g.Go(errgroupAssertSets(t, m2.domainManager, impl.SetDomainManager, impl.GetDomainManager))
+		g.Go(errgroupAssertSets(t, m2.taskManager, impl.SetTaskManager, impl.GetTaskManager))
+		g.Go(errgroupAssertSets(t, m2.visibilityManager, impl.SetVisibilityManager, impl.GetVisibilityManager))
+		g.Go(errgroupAssertSets(t, m2.replicationManager, impl.SetDomainReplicationQueueManager, impl.GetDomainReplicationQueueManager))
+		g.Go(errgroupAssertSets(t, m2.shardManager, impl.SetShardManager, impl.GetShardManager))
+		g.Go(errgroupAssertSets(t, m2.historyManager, impl.SetHistoryManager, impl.GetHistoryManager))
+		g.Go(errgroupAssertSets(t, m2.configManager, impl.SetConfigStoreManager, impl.GetConfigStoreManager))
+		require.NoError(t, g.Wait())
+		// execution managers are per shard, checked separately
+	})
+	t.Run("Execution manager getter", func(t *testing.T) {
```

Comment:
not related to this PR: why is this one treated differently?

Reply:
`GetExecutionManager` is the only one returning error because it supports lazy initialization. Other getters just return the member field `s.somethingManager`

Reply:
it also maintains a cached map of lazily-initialized ones, per shard.

so yea - different because it's very different.